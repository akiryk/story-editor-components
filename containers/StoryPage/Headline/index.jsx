/**
 * Story specific Headline stateless component.
 */
import React from 'react'

const Headline = ({ value, onChangeHandler, hasError }) => (
  <div className={`form-group ${hasError ? 'has-danger' : ''}`}>
    <input
      type="text"
      placeholder="Write a headline"
      className={`form-control story-headline ${hasError ? 'placeholder-danger' : ''}`}
      onChange={onChangeHandler}
      value={value}
    />
  </div>
)

Headline.propTypes = {
  onChangeHandler: React.PropTypes.func.isRequired,
  value: React.PropTypes.string,
  hasError: React.PropTypes.bool,
}

export default Headline

