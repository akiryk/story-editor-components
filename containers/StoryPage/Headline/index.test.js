import React from 'react'
import ReactTestRenderer from 'react-test-renderer'

import Headline from './index'


describe('StoryPage component', () => {
  it('renders Headline correctly', () => {
    const tree = ReactTestRenderer.create(
      <Headline
        value="This is a headline"
        onChangeHandler={(x) => (2 * x)}
        hasError={false}
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
