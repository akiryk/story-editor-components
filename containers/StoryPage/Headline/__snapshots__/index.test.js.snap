exports[`StoryPage component renders Headline correctly 1`] = `
<div
  className="form-group ">
  <input
    className="form-control story-headline "
    onChange={[Function]}
    placeholder="Write a headline"
    type="text"
    value="This is a headline" />
</div>
`;
